### PR TITLE
PRESIDECMS-608 One-to-many configurator form control

### PR DESCRIPTION
**Pull request by sebduggan**

### CODE REVIEW

File: system/handlers/formcontrols/OneToManyConfigurator.cfc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		var targetFk      = args.relationshipKey ?: args.sourceObject;
+		var useVersioning = Val( rc.version ?: "" ) && presideObjectService.objectIsVersioned( targetObject );
+		var hasSortOrder  = presideObjectService.getObjectProperties( targetObject ).keyExists( "sort_order" );
+		var orderBy       = hasSortOrder ? "sort_order" : "";
+		
+		if ( Len( Trim( args.savedData.id ?: "" ) ) ) {
+			var records       = presideObjectService.selectData(
+				  objectName       = targetObject
+				, filter           = { "#targetFk#"=args.savedData.id }
+				, selectFields     = [ "id", "label" ]
```
Comment (DominicWatson) [inline system/handlers/formcontrols/OneToManyConfigurator.cfc]: what happens here when the target object uses a label field other than "label", or doesn't have a label field at all? We _can_ use:

```
[ "id", "${labelfield} as label" ]
```

But perhaps it makes sense just to select all fields by not supplying a select fields array at all?

File: system/assets/js/admin/specific/datamanager/configuratorForm/configuratorFormHandler.js [hunk 2]
```diff
@@ -14,25 +14,9 @@
...
-						if ( inputValues[ i ] == item.value ) {
-							return item.text;
-						}
-					}
-				} ).get().join( ', ' );
-			});
-
+			// $configuratorForm.find(':checkbox:not(:checked)').attr('value', '0').prop('checked', true);
+			var formData = $configuratorForm.serializeObject();
+			console.log(formData);
```
Comment (DominicWatson) [inline system/assets/js/admin/specific/datamanager/configuratorForm/configuratorFormHandler.js]: Some debugging left over ;)

File: system/services/presideObjects/PresideObjectService.cfc [hunk 3]
```diff
@@ -1081,6 +1083,14 @@ component displayName="Preside Object Service" {
 		);
 
 		for (var record in records ) {
+			var formName = record.configurator__formname ?: "";
```
Comment (DominicWatson) [inline system/services/presideObjects/PresideObjectService.cfc]: I'd like to remove this logic if possible. Is the purpose to ensure that checkbox boolean fields have a present variable? I think this logic is the responsibility of the code that prepares the json (i.e. all the required fields should be present and its not the responsibility of the preside object service to start validating the input).

I think this should be safe to remove altogether (along with the line that gets the forms service). Or is this logic absolutely required?

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Looks great but for the forms service injection which we can discuss later if needed (though I think is safe to remove).

Also a bit of debugging left over in the js :)

File: system/assets/js/admin/specific/datamanager/configuratorForm/configuratorFormHandler.js (hunk 2 above)
Comment (sebduggan) [reply system/assets/js/admin/specific/datamanager/configuratorForm/configuratorFormHandler.js]: Oops...

File: system/services/presideObjects/PresideObjectService.cfc (hunk 3 above)
Comment (sebduggan) [reply system/services/presideObjects/PresideObjectService.cfc]: You're right, it was just there to ensure unchecked checkboxes were recorded. Now implemented in the Javascript.

**Comment by DominicWatson** [comment]
Nice one, thanks @sebduggan. Merging into upcoming 10.8 :)

**Pull request merged**